src/fileSyncer.ts

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as utilities from './utilities';
+import * as azHelper from './azureStorageHelper';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { SSHServer } from './interfaces';
+import * as sshHelper from './sshRunner';
+
+export enum TargetType {
+    cloudshell = 'cloudshell',
+    remotehost = 'remotehost'
+}
+
+export class FileSyncer {
```

Comment:
I think FolderSyncer or WorkspaceSyncer would be better

Reply:
renamed